File: code/0015/15.二维数组中的查找.cpp
class Solution {
public:
  bool searchArray(vector<vector<int>> array, int target) {
    if (array.empty())
      return false;
    int n = array.size(), m = array[0].size();
    for (int i = 0, j = m - 1; i < n && j >= 0;) {
      if (array[i][j] > target)
        j--;
      else if (array[i][j] < target)
        i++;
      else
        return true;
    }

    return false;
  }
};

File: code/0015/15.二维数组中的查找.py
class Solution(object):
    def searchArray(self, array, target):
        """
        :type array: List[List[int]]
        :type target: int
        :rtype: bool
        """
        if not array:
            return False
        
        n, m = len(array), len(array[0])
        i, j = 0, m - 1
        while i < n and j >= 0:
            if array[i][j] > target:
                j -= 1
            elif array[i][j] < target:
                i += 1
            else:
                return True
        
        return False
